docs/performance/inserts/methods.txt

```diff
@@ -0,0 +1,283 @@
+<insert_statement_types>`.
+
+.. _inserts_standard_querying:
+
+Standard Querying
+-----------------
+
+The standard way of issuing insert statements executes one statement at a time
+and does not make use of :ref:`inserts_bulk_operations` or any special
+:ref:`inserts_prepared_statements` client feature.
+
+For example, using the CrateDB Python client, here's a :ref:`single insert
+<inserts_single_inserts>`:
+
+.. code-block:: python
+
+   client.execute("INSERT INTO my_table (column_a) VALUES (?)", ["value 1"])
+
+.. _inserts_bulk_operations:
+
+Bulk Operations
+---------------
+
+You can use the `bulk operations`_ feature of the CrateDB HTTP endpoint to
+perform many inserts in a single operation.
+
+The advantages are the same as using the :ref:`UNNEST method<inserts_unnest>`:
+
+- Significantly less internal network traffic than executing each insert
+  statement individually
+
+- Even though you're executing multiple insert statements, the bulk query only
+  needs to be parsed, planned, and executed once
+
+- If `translog.durability`_ is set to ``REQUEST`` (the default), a bulk insert
+  only flushes the disk once
+
+Because the advantages are the same as using the ``UNNEST`` method, you
+typically will not see a performance improvement by combining bulk operations
+with ``UNNEST`` statements or statements with :ref:`multiple value expressions
+<inserts_multiple_values>`.
+
+Bulk operations are typically done with :ref:`single insert statements
+<inserts_single_inserts>` as an alternative to the ``UNNEST`` method.
+
+.. _inserts_prepared_statements:
+
+Prepared Statements
+-------------------
+
+Some clients offer a prepared statements feature. Prepared statements are
+parsed by CrateDB and can then be executed any number of times without having
+to re-parse.
+
+This functionality is often presented as batch execution. `The JDBC client`_,
+for instance, provides the `addBatch`_ and `executeBatch`_ methods.
+
+For example:
+
+.. code-block:: java
+
+   PreparedStatement preparedStatement = connection.prepareStatement(
+       "INSERT INTO my_table (id, first_name) VALUES (?, ?)");
+
+   preparedStatement.setString(1, "Arthur");
+   preparedStatement.addBatch();
+
+   preparedStatement.setString(1, "Trillian");
+   preparedStatement.addBatch();
+
+   preparedStatement.setString(1, "Marvin");
+   preparedStatement.addBatch();
+
+   int[] results = preparedStatement.executeBatch();
+
+In addition to reducing parsing overhead, prepared statement execution requests
+use the binary protocol, contain almost no headers, and are executed over an
+already established connection.
+
+Typically, prepared statements are used :ref:`single insert statements
+<inserts_single_inserts>`.
+
+Prepared statements with single inserts will usually perform better than
+:ref:`standard querying <inserts_standard_querying>` with single inserts, but
+worse than standard querying with both the :ref:`UNNEST method<inserts_unnest>`
```

Comment:
It should be comparable to unnest/multi-value expression as we internally try to execute the whole batch as one, so you've the same benefits (1 flush per shard being written to for the whole batch).

Reply:
changed to:

> and should be comparable to standard querying with both